### PR TITLE
feat(billing): lastResetAt + archivedAt schema + meter cap (PR-N8)

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## Summary

P2 robustness pass identified by `/critical-review` after PR-N5.

- **P2.8 — `lastResetAt` on Subscription**: replaces the 7-day window query (which silently skipped subscriptions when the scheduler was delayed > 7d). New repository method `findAllDueForResetByLastReset(now)` returns active subscriptions where `lastResetAt is null OR lastResetAt < now - 7 days`. `resetAllDue` now persists `lastResetAt` after each successful `resetWeek`. Legacy `findAllDueForReset` kept and marked `@deprecated`.
- **P2.9 — `archivedAt` declared**: `archivedAt: Date | null` is now in both Mongoose and Zod schemas of `BillingUsage`. `archiveOtherWeeks()` uses `archivedAt: null` instead of `$exists: false` so the query matches the schema default. Without this, writes to `archivedAt` were silently stripped per Mongoose strict mode (cf `feedback_mongoose_unset_strict`).
- **P2.12 — `maxUnitsPerOperation` cap**: meter `attribute()` clamps `units` to `maxUnitsPerOperation` (with `console.warn` when clamping kicks in) before calling `incrementMeter`. Feature breakdown rescaled proportionally so persisted buckets sum to `meterUsed`.

## Test plan

- [x] `npm run lint` — clean
- [x] `npm run test:unit -- billing` — 973/973 passing
- [x] 8 new unit tests across `billing.reset.service`, `billing.subscription.repository`, `billing.usage.repository`, `billing.meter.service`
- [x] Updated `scripts/tests/billing.cron.weeklyReset.unit.tests.js` with new mock surface (`findAllDueForResetByLastReset`, `updateLastResetAt`)

## Notes

- No data migration: `lastResetAt` defaults to `null`, existing subscriptions are picked up by the new query naturally.
- No new file added — all changes in existing models/services/tests.

### CODE REVIEW

File: modules/billing/services/billing.meter.service.js [hunk 1]
```diff
@@ -56,6 +56,33 @@ const unitsFromCosts = async (costs, planId, ratioVersion) => {
...
+
+  const cappedBreakdown = {};
+  let remaining = cappedUnits;
+
+  for (const [key, value] of Object.entries(breakdown)) {
+    if (!Number.isFinite(value) || value <= 0 || remaining <= 0) continue;
+
+    const applied = Math.min(value, remaining);
+    if (applied > 0) {
+      cappedBreakdown[key] = applied;
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/billing/services/billing.meter.service.js]: Fixed in 960b9e9f: replaced `{}` with `Object.create(null)` in `capBreakdown` to eliminate prototype pollution risk.

File: modules/billing/services/billing.meter.service.js [hunk 2]
```diff
@@ -56,6 +56,33 @@ const unitsFromCosts = async (costs, planId, ratioVersion) => {
...
+      remaining -= applied;
+    }
+  }
+
+  return cappedBreakdown;
+};
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/billing/services/billing.meter.service.js]: Fixed in 960b9e9f: `capBreakdown` now uses proportional rescaling (`floor(value * cappedUnits / originalTotal)`) with remainder distribution to the largest buckets first, ensuring sum(result) === cappedUnits exactly. `originalTotal` is passed as a third argument from `attribute()`.

File: modules/billing/services/billing.meter.service.js [hunk 3]
```diff
@@ -90,12 +117,21 @@ const attribute = async (history, organizationId) => {
     ({ totalUnits, breakdown } = await unitsFromCosts(history.costs, planId, ratioVersion));
   }
 
+  const maxUnits = config?.billing?.meter?.maxUnitsPerOperation ?? Infinity;
+  const cappedUnits = Math.min(totalUnits, maxUnits);
+  const cappedBreakdown = cappedUnits < totalUnits ? capBreakdown(breakdown, cappedUnits) : breakdown;
+  if (cappedUnits < totalUnits) {
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/billing/services/billing.meter.service.js]: Fixed in 960b9e9f: extracted `const isCapped = cappedUnits < totalUnits` and reused it for both the breakdown adjustment and the console.warn — eliminating the duplicate comparison.

File: modules/billing/repositories/billing.subscription.repository.js [hunk 4]
```diff
@@ -128,6 +129,52 @@ const findAllDueForReset = (from, to) =>
...
+  const threshold = new Date(now.getTime() - 7 * 24 * 60 * 60 * 1000);
+
+  return Subscription.find(
+    {
+      status: { $in: ['active', 'trialing'] },
+      currentPeriodStart: { $ne: null },
+      $or: [
+        { lastResetAt: null },
+        { lastResetAt: { $lt: threshold } },
+      ],
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/billing/repositories/billing.subscription.repository.js:167]: In MongoDB, a missing field satisfies `{ field: null }` (equality to null), so `{ field: { $ne: null } }` correctly excludes both null-valued *and* missing fields — this is documented MongoDB behavior. Subscriptions without a `currentPeriodStart` field in the DB will *not* be matched. No code change needed.

File: modules/billing/services/billing.meter.service.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/billing/services/billing.meter.service.js:100]: Fixed in 960b9e9f: `capBreakdown` now uses proportional rescaling (`floor(value * cappedUnits / originalTotal)`) with remainder distributed to the largest buckets first, so sum(result) === cappedUnits and all proportions are preserved. 3 new unit tests cover the multi-key proportional case, the floor-remainder distribution, and the empty-breakdown edge case.
